src/app/services/posts.service.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Subject } from 'rxjs';
+
+
+const baseURL = 'https://jsonplaceholder.typicode.com/';
```

Comment:
this should be a class property, no need to extract it separately... you can also consider putting it in environment and than in class baseUrl = environment.apiUrl